### PR TITLE
MINOR: improve JavaDocs for KafkaStreams, KafkaProducer, KafkaConsumer

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @dguy @bbejeck @hachikuji @apurvam @hjafarpour @rodesai

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java [hunk 1]
```diff
@@ -579,7 +579,9 @@
      * either strings or objects of the appropriate type (for example a numeric configuration would accept either the
      * string "42" or the integer 42).
      * <p>
-     * Valid configuration strings are documented at {@link ConsumerConfig}
+     * Valid configuration strings are documented at {@link ConsumerConfig}.
+     * <p>
+     * Note: after using the {@code KafkaConsumer} you must always {@link #close()} it to avoid resource leaks.
```
Comment (apurvam) [inline clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: I wonder if it makes sense to simplify this and say that 'after creating a {@code KafkaConsumer} you must always call close'. 

This is not strictly true, but would make it consistent with the streams and producer javadoc. If we want to be specific, then we should go the whole way and say 'after calling poll, we should always close'.

**Review by apurvam — COMMENTED** [review]
The producer and streams java doc looks good. 

I think we should think about the wording about the kafka consumer java doc though. As it stands, it seems a bit ambiguous and less useful than the other two.

**Comment by mjsax** [comment]
Updates this. I am fine with the simple way.

**Comment by mjsax** [comment]
Retest this please

**Review by apurvam — APPROVED** [review]
LGTM! thanks! The test failure definitely seems unrelated.

**Comment by mjsax** [comment]
Retest this please. (I looked at the test -- it has a race condition...)

**Comment by guozhangwang** [comment]
LGTM! Merged to trunk and 1.0.

**Pull request closed without merging**